### PR TITLE
DQM framework: header changes

### PR DESCRIPTION
**Pull request by schneiml**
#### PR description:

This PR contains two things:
- A lot of automated changes to the `#include`'s in DQM code, to not explicitly refer to soon-to-be-remove or redundant header files. This is not all complete, but it should keep the future PRs smaller and avoid conflicts. There are some manual changes required, which is why I'd prefer to not do this migration again.
- A set of automated refactoring tools, that I included simply because this is the first PR made using them [1]. They don't really need to go into any release, but might be useful to others (also makes my life easier). (Update: moved to https://github.com/cms-sw/cmssw/compare/master...schneiml:dqm-refactoring-tools for now.)

Eventually, there will be a PR integrating the changes produced by the other tools, but for that to happen, we first need to implement the required code in `DQMServices/Core`...

I am wondering how we should move forward with the formatting changes, I could (might be required to?) include a large batch of them here.
 
#### PR validation:

None so far, but things compile, and none of the changes should affect the behavior of the code.

I'd like to raise special attention to 38e9f49, which as far as I see, should not be required, yet, without it, I got a compilation error. Could this be a bug/incorrect use of scram (@smuzaffar ?). I am not sure what the risks of linking to another package in Fastsim are.

[1] Actually, this is not true, #26495 also used a tool included here.

### CODE REVIEW

File: HLTrigger/Timer/plugins/FastTimerService.cc [hunk 1]
```diff
@@ -622,13 +621,13 @@ FastTimerService::PlotsPerPath::book(
   for (unsigned int bin: boost::irange(0u, bins)) {
     auto const& module = job[path.modules_and_dependencies_[bin]];
     std::string const& label = module.scheduled_ ? module.module_.moduleLabel() : module.module_.moduleLabel() + " (unscheduled)";
-    module_counter_          .setBinLabel(bin + 1, label.c_str());
-    module_time_thread_total_.setBinLabel(bin + 1, label.c_str());
-    module_time_real_total_  .setBinLabel(bin + 1, label.c_str());
```
Comment (Martin-Grunewald) [inline HLTrigger/Timer/plugins/FastTimerService.cc]: Why these changes?

File: HLTrigger/Timer/plugins/FastTimerService.cc [hunk 2]
```diff
@@ -622,13 +621,13 @@ FastTimerService::PlotsPerPath::book(
...
+    module_counter_          .setBinLabel(bin + 1, label);
+    module_time_thread_total_.setBinLabel(bin + 1, label);
+    module_time_real_total_  .setBinLabel(bin + 1, label);
```
Comment (Martin-Grunewald) [inline HLTrigger/Timer/plugins/FastTimerService.cc]: Why these changes?

**Comment by perrotta** [comment]
@cmsbuild please test

File: HLTrigger/Timer/plugins/FastTimerService.cc (hunk 2 above)
Comment (schneiml) [reply HLTrigger/Timer/plugins/FastTimerService.cc]: This should most likely not be here, but I am not sure where it comes from. Either one of the tools did something crazy or maybe there where commits to this in the last days. Or I screwed up manually. I'll have a look.

(Thanks for looking at this insane list of changes, btw!)

**Comment by schneiml** [comment]
@fabiocos seeing the amount of packages/signatures touched here makes me wonder if it would be better to split this into multiple PRs. I expected most changes to be under DQM responsibility...

Also, any comments regarding the interaction of `clang-format` PRs with this?

File: HLTrigger/Timer/plugins/FastTimerService.cc (hunk 2 above)
Comment (schneiml) [reply HLTrigger/Timer/plugins/FastTimerService.cc]: Ah, easy. It was the code checks. Not sure what exactly is going on there but apparently our code-checks think the `c_str` is not required (and I'd agree, if it compiles it is probably better like this).

File: HLTrigger/Timer/plugins/FastTimerService.cc (hunk 1 above)
Comment (schneiml) [reply HLTrigger/Timer/plugins/FastTimerService.cc]: Code checks, see other comment.

**Comment by schneiml** [comment]
please test

File: DQMServices/Components/scripts/apipopularity.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+(sqlite3 $1 | column -t -s'|') <<SQL
```
Comment (makortel) [inline DQMServices/Components/scripts/apipopularity.sh]: These files end up in everyones `$PATH`. Ok, they don't seem to have their execution bit set, and they're missing `#!` so one can't really execute them (without explicitly passing them to a shell), but to me most of the scripts have rather generic names (like `csv_to_sqlite.sh` or all the `*dxr*` ones) leading to risks of name collisions.

File: DQMServices/Components/scripts/dxr-index.cpp [hunk 4]
```diff
@@ -0,0 +1,1490 @@
+#include "clang/AST/AST.h"
```
Comment (makortel) [inline DQMServices/Components/scripts/dxr-index.cpp]: Having source code in `scripts` looks weird.

**Comment by fabiocos** [comment]
@schneiml I am going to integrate a good fraction of the already prepared clang-format migration for DQM, so not sue how much remains outside that could be blocked by this PR. But in principle this PR itself could provide a migration for not already moved packages. 

For what I can see most of the packages affected by this PR are under your responsibility, but still you touch many other areas..

**Comment by fabiocos** [comment]
@schneiml indeed this PR needs now to be rebased

File: DQMServices/Components/scripts/apipopularity.sh (hunk 3 above)
Comment (schneiml) [reply DQMServices/Components/scripts/apipopularity.sh]: Yep, I am aware. Any better place to hide such things without removing them entirely?

I guess most of them can be deleted in the end, but a few could remain useful.

File: DQMServices/Components/scripts/dxr-index.cpp (hunk 4 above)
Comment (schneiml) [reply DQMServices/Components/scripts/dxr-index.cpp]: Similar -- I just wanted some place to put the files under version control, but clearly this should not be compiled into CMSSW. Also, it does not really fit bin/, since actually that compiles into a library.

Getting access to the DXR index database in some official way might be useful, but for now, this pile of scripts has served me fairly well... (Afair scram can build the dxr index, but actually I didn't even bother to check what the output of that would be).

Comment (schneiml) [reply DQMServices/Components/scripts/dxr-index.cpp]: I put the tools onto a different branch for now. I'd still be happy about proposals for a more permanent place for such things.

**Comment by schneiml** [comment]
Rebased to CMSSW_11_X_2019-05-15-2300. I removed the tools for now. 

Trying to add the formatting changes as well now.

**Comment by schneiml** [comment]
Can we get https://github.com/cms-sw/cmssw/pull/26161 soonish or should I just include it here? that one would sure cause trouble with the code-format.

**Comment by schneiml** [comment]
Ah, this PR doesn't actually touch the `DQMStore` and `code-format` therefore also leaves it alone. So no conflict with  #26161 (at least nothing big). 

I added `scram b code-format`, which, as far as I see, only affected the packages that where modified anyways. So this should be good now?

**Comment by schneiml** [comment]
please test

Hope everything still compiles.

File: DQMServices/Components/scripts/dxr-index.cpp (hunk 4 above)
Comment (makortel) [reply DQMServices/Components/scripts/dxr-index.cpp]: I'm pretty sure someone will object, but `test`? (technically these are not unit tests, but we tend to have "random stuff" in `test` in many packages)

**Comment by slava77** [comment]
> Trying to add the formatting changes as well now.

this seems to overlap with a lot of centrally made PRs doing clang-format.
May I propose to postopone this PR until after the migration is done centrally.

**Comment by schneiml** [comment]
@slava77 Yes, esp. since `scram b code-format` only formats files with changes, so even though I touch lots of packages they are not fully formatted in the end? 

I wanted to get that in and have a fairly stable base to work on with 11, but it seems easier now to have my branch(es) diverge for another week or two and wait for the code-format wave to calm down.

**Comment by schneiml** [comment]
Will come back later.
